amethyst_animation/src/resources.rs

```diff
@@ -737,13 +737,14 @@ where
     ///
     /// This should be used with care, as this will leave all linked samplers in place. If in
     /// doubt, use `abort()` instead.
-    pub fn remove(&mut self, id: I) {
+    pub fn remove(&mut self, id: I) -> &mut Self {
```

Comment:
Does it make sense to have these consume and return `self` rather than a reference to it?
`pub fn remove(self, id: I) -> Self`. If we expect it to work like a builder it's a bit more common.

Reply:
Would not work when pulled from a resource that is inside of `World` already.

Reply:
ah fair. Should have looked deeper into where it's called from 👍